src/main.cpp

```diff
@@ -2636,9 +2636,11 @@ bool ActivateBestChain(CValidationState &state, const CChainParams& chainparams,
             // Notify external listeners about the new tip.
             if (!vHashes.empty()) {
                 GetMainSignals().UpdatedBlockTip(pindexNewTip);
-                uiInterface.NotifyBlockTip(vHashes.front());
             }
         }
+        if (!vHashes.empty()) {
+            uiInterface.NotifyBlockTip(fInitialDownload, pindexNewTip);
+        }
```

Comment:
@sdaftuar: I'm not familiar with the block announcements with headers, could you check if this rebase makes sense?